src/NServiceBus.Core/Transports/Msmq/MsmqMessageDispatcher.cs

```diff
@@ -129,12 +129,27 @@ bool IsCombiningTimeToBeReceivedWithTransactions(ReadOnlyContextBag context, Dis
                 return false;
             }
 
+            if (Transaction.Current != null)
```

Comment:
Agree with @SzymonPobiega putting `Transaction.Current` into `TransportTransaction` is better than accessing ambient `Transaction.Current`. But what are the implications when a dispatcher is called outside a receive pipeline and a user is surrounding with a `TransactionScope`? We need to make sure that we put then the `transaction.Current` into the bag as well.